### PR TITLE
Beam - fixed inconsistencies of draw method

### LINKED ISSUE #24043: Beam - inconsistent convention used on draw()
The `draw()` method of the `Beam` class is used to plot a schematic view of the loads applied on the beam. I mainly use it to verify that I have applied the loads where they should. It doesn't compute anything. However, I came to the realization that it is inconsistent with the sign convention used by the class, which might leads to confusion. 

This is what I understood by reading [the documentation](https://docs.sympy.org/latest/modules/physics/continuum_mechanics/beam.html?highlight=beam#sympy.physics.continuum_mechanics.beam.Beam) and looking at the symbolic results. For the purpose of computation, positive loads are directed downward, negative loads are directed upwards. Symbolic loads (such as support reactions) are assumed positive, hence directed downwards.

Instead, the `draw()` method always plots symbolic reactions and loads directed towards the top.

A few simple examples should clarify the situation.

```python
from sympy import *
from sympy.plotting import PlotGrid
from sympy.physics.continuum_mechanics import Beam
init_printing()

E, I, P = symbols('E, I, P')
R1, R2 = symbols('R1, R2')
L = symbols("L", positive=True)

b1 = Beam(L, E, I)
b1.apply_load(R1, 0, -1)
b1.apply_load(R2, L, -1)
b1.apply_load(5, L/4, -1)
b1.bc_deflection = [(0, 0), (L, 0)]
b1.solve_for_reaction_loads(R1, R2)
p1 = b1.draw()

b2 = Beam(L, E, I)
b2.apply_load(R1, 0, -1)
b2.apply_load(R2, L, -1)
b2.apply_load(-5, L/4, -1)
b2.bc_deflection = [(0, 0), (L, 0)]
b2.solve_for_reaction_loads(R1, R2)
p2 = b2.draw()

b3 = Beam(L, E, I)
b3.apply_load(R1, 0, -1)
b3.apply_load(R2, L, -1)
b3.apply_load(P, L/4, -1)
b3.apply_load(5, L/2, -1)
b3.bc_deflection = [(0, 0), (L, 0)]
b3.solve_for_reaction_loads(R1, R2)
p3 = b3.draw()

p1.title = "W = +5 -> " + str(b1.reaction_loads)
p2.title = "W = -5 -> " + str(b2.reaction_loads)
p3.title = "W = P + 5 -> " + str(b3.reaction_loads)
PlotGrid(2, 2, p1, p2, p3)
```

index

However, let's look at the results:
* top-left plot with positive numeric load. Symbolic results are computed with the above convention (positive loads downward, symbolic loads downward): in words, reaction loads have opposite direction than the load. Intuitively, the plot appears correct, but this is just by chance as the `draw()` method doesn't compute anything. A user comparing symbolic results and the schematic view might think that the reaction loads are directed in the opposite direction that what it is currently drawn, which would be wrong. The correct plot would have the reaction loads pointing downward: the user then looks at the symbolic results, see that they must be directed opposite to the numeric load, so in reality the reaction loads would be directed upward. Essentially, if I use the current schematic to perform computations by hand, I would get reaction loads with opposite sign than the ones computed by SymPy.
* top-right plot with negative numeric load. As the (symbolic) reaction loads are positive, a user might get the impression that there is agreement between what is computed by SymPy and the direction illustrated in the schematic. This is wrong. Again, by doing computations by hand based on the schematics, I'll get opposite signs on the reactions than what computed by SymPy.
* Bottom-left plot: positive numeric load and symbolic load. The `Beam` class consider both loads directed downward. The `draw()` method consider the symbolic load directed upward. This results in a conceptual mess. If we do computations by hand based on the schematic, we'll get different results than SymPy.

I believe that the `draw()` method should use the same convention as used by the class. Sure, reaction loads will be pointed downwards in the schematic view, which is pretty much against what we usually use in hand written computations. But at least this will bring coherence with the symbolic results.

I'm probably going to add a few more inconsistencies on this issue about the `draw()` method later on.

### PR DESCRIPTION
**Pull request by Davide-sd**
#### References to other Issues or PRs

Fixes #24043 

#### Brief description of what is fixed or changed

This PR is fixing the inconsistencies mentioned on Issue #24043 , thus providing a schematic view which is consistent with the sign convention used by the `Beam` class. Not only this improves the visualization. It also allows the users to better make sense of the symbolic results. In particular:

* the `draw` method is no longer looking at the assumptions on symbolic loads and moments, which are not used by the `Beam` class either in its computations.
* positive loads (numeric or symbolic) are pointing downward.
* negative loads (numeric or symbolic) are pointing upward.
* positive moments (numeric or symbolic) are directed counterclockwise.
* negative moments (numeric or symbolic) are directed clockwise.
* the `draw` method is now able to visualize symbolic distributed loads without raising errors.
* the beam won't be hidden by the distributed loads.

#### Release Notes

* physics.continuum_mechanics
  * Improved the schematic view produced by the `Beam.draw` method.

### CODE REVIEW

**Comment by moorepants** [comment]
@AdvaitPote, given that you are also working on draw methods for the trusses, it would be in your best interest to review this PR and discuss.

File: sympy/physics/continuum_mechanics/beam.py [hunk 1]
```diff
@@ -2189,13 +2214,64 @@ def draw(self, pictorial=True):
...
+
+        Returns
+        =======
+        1: if the load is negative
+        0: if the load is positive
+        -1: if it is indeterminate
+
+        Notes
+        =====
+        The Beam class doesn't care about assumptions on symbolic loads.
```
Comment (moorepants) [inline sympy/physics/continuum_mechanics/beam.py:2253]: The Beam class *should* care about assumptions. That is the only reliable way to know if a symbol or expression involving symbols is positive or negative.

**Comment by oscarbenjamin** [comment]
This has merge conflicts now. I'll mark this as draft. Feel free to mark it as ready for review later.

File: sympy/physics/continuum_mechanics/beam.py [hunk 2]
```diff
@@ -2213,95 +2299,88 @@ def _draw_load(self, pictorial, length, l):
...
             # point loads
             if load[2] == -1:
-                if isinstance(load[0], Symbol) or load[0].is_negative:
-                    annotations.append({'text':'', 'xy':(pos, 0), 'xytext':(pos, height - 4*height), 'arrowprops':{"width": 1.5, "headlength": 5, "headwidth": 5, "facecolor": 'black'}})
+                iln = self._is_load_negative(load[0])
+                if iln == -1:
+                    warning_body += "* Point load %s located at %s\n" % (load[0], load[1])
+
+                if iln:
+                    annotations.append({'text':'', 'xy':(pos, 0), 'xytext':(pos, height - 4*height), 'arrowprops':dict(width= 1.5, headlength=5, headwidth=5, facecolor='black')})
```
Comment (smichr) [inline sympy/physics/continuum_mechanics/beam.py:2307]:
```suggestion
                    annotations.append({'text':'', 'xy':(pos, 0), 'xytext':(pos, height - 4*height), 'arrowprops':{'width': 1.5, 'headlength': 5, 'headwidth': 5, 'facecolor': 'black'}})
```

File: sympy/physics/continuum_mechanics/beam.py (hunk 1 above)
Comment (smichr) [reply sympy/physics/continuum_mechanics/beam.py]: It now cares and also tries doit and expand for non-atomic expressions if the give None in their raw state.

**Comment by smichr** [comment]
@AdvaitPote @moorepants , please give this another look.

**Comment by smichr** [comment]
Further modifications can be made in a new PR.

**Pull request merged**